src/test/java/main/HttpMethodsTest.java

```diff
@@ -0,0 +1,151 @@
+package main;
+
+import io.restassured.RestAssured;
+import io.restassured.http.Method;
+import io.restassured.response.Response;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.HashMap;
+
+import static io.restassured.RestAssured.given;
+import static org.testng.Assert.assertEquals;
+
+public class HttpMethodsTest {
```

Comment:
I would approach this test differently. In data provider I would put { "HTTP METHOD NAME",  "EXPECTED STATUS CODE") something like:
 Object[][] dataParameters = {
       {"GET", 200},
       {"POST", 405},
       {"PUT", 405},
       {"HEAD", 4XX},
.... etc
     };
and then you will have one one test method in which based on the HTTP Method name passed as argument you can send a proper type of request. To choose proper type you can use e.g. switch statement.